src/parser/jobs/mch/modules/Wildfire.tsx

```diff
@@ -0,0 +1,204 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import {ActionLink} from 'components/ui/DbLink'
+import {RotationEvent} from 'components/ui/Rotation'
+import {RotationTable} from 'components/ui/RotationTable'
+import {Event, Events} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {EventHook} from 'parser/core/Dispatcher'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {History, HistoryEntry} from 'parser/core/modules/ActionWindow/History'
+import {Data} from 'parser/core/modules/Data'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import {Timeline} from 'parser/core/modules/Timeline'
+import React, {Fragment} from 'react'
+import {Message} from 'semantic-ui-react'
+
+// We always want 6 GCDs in WF
+const EXPECTED_GCDS = 6
+
+const SEVERITIES = {
+	BAD_WILDFIRE: {
+		1: SEVERITY.MINOR,
+		2: SEVERITY.MEDIUM,
+		3: SEVERITY.MAJOR,
+	},
+	FIZZLED_WILDFIRE: {
+		1: SEVERITY.MEDIUM,
+		2: SEVERITY.MAJOR,
+	},
+}
+
+interface WildfireWindow {
+	actionEvents: Array<Events['action']>
+	stacks: number
+	damage?: number
+}
+
+export class Wildfire extends Analyser {
+	static override handle = 'wildfire'
+	static override title = t('mch.wildfire.title')`Wildfire`
+
+	@dependency private data!: Data
+	@dependency private suggestions!: Suggestions
+	@dependency private timeline!: Timeline
+
+	private history = new History<WildfireWindow>(
+		() => ({
+			actionEvents: [],
+			stacks: 0,
+		})
+	)
+
+	private actionHook?: EventHook<Events['action']>
+
+	private actionFilter = filter<Event>()
+		.source(this.parser.actor.id)
+		.type('action')
+
+	override initialise() {
+		this.addEventHook(filter<Event>()
```

Comment:
Minor nit but it'd be a little cleaner to do a `const playerFilter = filter<Event>().source(this.parser.actor.id)` and then reuse that in each of the 3 hooks.

Reply:
👍